Export for curseforge modpacks

TOLoneWolf opened the pull request:
- Defaults to Minecraft player name for author.
- Defaults to instance name for export name.

TOLoneWolf committed:
Create Curseforge Exporter

TOLoneWolf committed:
Fix indentation

TOLoneWolf committed:
fix

TOLoneWolf committed:
change

TOLoneWolf committed:
Merge branch 'next' into exportTwitchPack

TOLoneWolf committed:
Some fixes.

TOLoneWolf committed:
clean up

TOLoneWolf committed:
Fix linux path for vanilla launcher launcher_profiles.json

TOLoneWolf committed:
Remove a bunch of commeneted code bits.
Change some names to better describe what they are.

TOLoneWolf committed:
change

blarfoon commented on src/app/desktop/components/Instances/Instance.js:
```diff
@@ -336,6 +340,20 @@ const Instance = ({ instanceName }) => {
           Open Folder
         </MenuItem>
+
+        {/* // TODO - Support other export options besides curseforge forge. */}
```
We could actually add a dropdown menu inside the export modal to choose the format to export to

blarfoon commented on src/common/modals/InstanceExport/BackButton.js:
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Button } from 'antd';
+import styles from './ExportPackModal.module.css';
```
Do not use external css modules. Use styled components

blarfoon commented on src/common/modals/InstanceExport/BackButton.js:
```diff
@@ -0,0 +1,16 @@
+export default function BackButton({ onClick, disabled = false }) {
+  return (
+    <Button
+      className={styles.backBtn}
```
Do not use external css modules. Use styled components

blarfoon commented on src/common/modals/InstanceExport/FirstStep.js:
```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import path from 'path';
+import { Button, Input } from 'antd';
+import styles from './ExportPackModal.module.css';
```
Avoid css modules

blarfoon commented on src/common/modals/InstanceExport/FirstStep.js:
```diff
@@ -0,0 +1,102 @@
+          <Button type="primary" onClick={showFileDialog}>
+            {
+              // eslint-disable-next-line no-nested-ternary
+              filePath === null
```
this can be written as `!filePath`

blarfoon commented on src/common/modals/InstanceExport/FirstStep.js:
```diff
@@ -0,0 +1,102 @@
+      <ContinueButton
+        onClick={setPage}
+        disabled={
+          !(packZipName && packVersion && packAuthor && filePath !== null)
```
`filePath !== null` can be written as `filePath`

blarfoon commented on src/common/modals/InstanceExport/SecondStep.js:
```diff
@@ -0,0 +1,95 @@
+      const arr = dirTree(instancePath);
+
+      const mapArr = (children, disableChildren = false) => {
+        if (children === undefined || children.length === 0) return [];
```
`children === undefined` = `!children`

blarfoon commented on src/common/modals/InstanceExport/ThirdStep.js:
```diff
@@ -0,0 +1,153 @@
+      );
+
+      // Process files from selection
+      await fse.mkdirp(path.join(tempExport, 'overrides'));
```
use `makeDir` from `make-dir`

blarfoon commented on src/common/modals/InstanceExport/ThirdStep.js:
```diff
@@ -0,0 +1,153 @@
+            );
+          }
+        }),
+        { concurrency: 30 }
```
use 10, 30 might be blocking the Ui thread. Also, concurrency does not work on Promise.all. Use pMap if you want concurrency

blarfoon commented on src/common/modals/InstanceExport/ThirdStep.js:
```diff
@@ -0,0 +1,153 @@
+          <div>
+            <h2>
+              All Done!{' '}
+              <FontAwesomeIcon icon={faCheck} style={{ color: '#27ae60' }} />
```
Don't use `style` and do not hardcode colors

blarfoon commented on src/common/reducers/actions.js:
```diff
@@ -422,7 +422,10 @@ export function loginThroughNativeLauncher() {
-    const vanillaMCPath = path.join(homedir, mcFolder);
+    const vanillaMCPath =
+      process.platform === 'linux'
+        ? path.join(homedir, '../', mcFolder)
```
use path.resolve here

blarfoon commented on src/common/modals/InstanceExport/ThirdStep.js:
```diff
@@ -0,0 +1,153 @@
+        version: modloader[1],
+        modLoaders: [
+          {
+            id: `forge-${forgeVersion}`,
```
Don't hardcode `forge` because we also want to support fabric exports

TOLoneWolf replied:
I have to take a look if your import is even capable of reading anything other then forge ffrom curseforge manifest format.

Note: This exporter is specifically coded for curseforge and the only thing they ever support is forge.  The  output format might be different if fabric.

blarfoon commented on src/common/modals/InstanceExport/ThirdStep.js:
```diff
@@ -0,0 +1,153 @@
+      files: mods.map(mod => ({
+        projectID: mod.projectID,
+        fileID: mod.fileID,
+        required: true
```
Care that not all mods have the fileId and projectId, we want to filter them before adding them here

blarfoon reviewed (changes requested):
Generally speaking this is good. I didn't understand exactly how mods are handled. Mods should be excluded from selection and mods that are not on cf (hence don't have projectId and fileId) should be automatically added to overrides.

TOLoneWolf committed:
Fixes

TOLoneWolf committed:
Moved things around.
- Selection now applies to curseforge mods listed in the config of the instance. Where always included before.
- All tree nodes selected by default.

TOLoneWolf committed:
Limit input

TOLoneWolf committed:
HardLink files temp folder for packing in export. Fails over to standard copy.

TOLoneWolf committed:
Removed default version 0.1.0 to just blank.

TOLoneWolf committed:
Merge branch 'next' into exportTwitchPack

TOLoneWolf committed:
blacklist manifest.json file from export.

TOLoneWolf committed:
Move from InstanceExport to CurseForge sub folder. This is for when more export options are added later.

TOLoneWolf committed:
changed names to remove underscores.

blarfoon commented on src/common/modals/InstanceExport/CurseForge/ContinueButton.js:
```diff
@@ -0,0 +1,36 @@
+export default function ContinueButton({ onClick, disabled = false }) {
+  return (
+    <div
+      onClick={!disabled ? () => onClick(s => s + 1) : null}
```
`onClick={!disabled ? () => onClick(s => s + 1) : null}` write this as
```
onClick={() => {
  if (!disabled) {
    onClick(s => s + 1)
  }
}}
```

blarfoon commented on src/common/modals/InstanceExport/CurseForge/FirstStep.js:
```diff
@@ -0,0 +1,281 @@
+                ? ''
+                : filePath.length >= 80
+                ? `...${filePath.slice(-80)}`
+                : filePath
```
Don't use nested ternary, they are unreadable

TOLoneWolf committed:
Fix Buttons

TOLoneWolf committed:
ternary to function

TOLoneWolf committed:
oops
